pandas/_libs/dtypes.pxd

```diff
@@ -5,13 +5,40 @@ Common location for shared fused types
 from numpy cimport (
     float32_t,
     float64_t,
+    int8_t,
+    int16_t,
+    int32_t,
     int64_t,
+    uint8_t,
+    uint16_t,
+    uint32_t,
     uint64_t,
 )
 
+ctypedef fused numeric_t:
```

Comment:
can you add a comment before each one which explains a high-level what (e.g. int + uint + float)

Reply:
Added